shell/platform/darwin/ios/framework/Source/FlutterObservatoryPublisher.mm

```diff
@@ -233,9 +199,15 @@ - (instancetype)init {
        runner = fml::MessageLoop::GetCurrent().GetTaskRunner()](const std::string& uri) {
         if (!uri.empty()) {
           runner->PostTask([weak, uri]() {
+            // uri comes in as something like 'http://127.0.0.1:XXXXX/' where XXXXX is the port
+            // number.
             if (weak) {
-              [[weak.get() delegate]
-                  publishServiceProtocolPort:[NSString stringWithUTF8String:uri.c_str()]];
+              NSURL* url =
+                  [[NSURL alloc] initWithString:[NSString stringWithUTF8String:uri.c_str()]];
+              weak.get().url = url;
```

Comment:
@gaaclarke What's the engine idiomatic way to grab a strong pointer to retain this in scope, instead of getting the weak pointer each time?

Reply:
I don't believe there is one.  The weakptr's are limited to being used one a single thread so there is no risk that it will get deleted or edited in this scope.